backend/tracim_backend/applications/share/models.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+import enum
+from hashlib import sha256
+import os
+import typing
+
+from sqlalchemy import Boolean
+from sqlalchemy import Column
+from sqlalchemy import DateTime
+from sqlalchemy import Enum
+from sqlalchemy import ForeignKey
+from sqlalchemy import Integer
+from sqlalchemy import Sequence
+from sqlalchemy import Unicode
+from sqlalchemy.orm import relationship
+from sqlalchemy.orm import synonym
+
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import Content
+from tracim_backend.models.meta import DeclarativeBase
+
+
+class ContentShareType(enum.Enum):
+    EMAIL = "email"
+    PUBLIC_LINK = "public-link"
+
+
+class ContentShare(DeclarativeBase):
+
+    MIN_PASSWORD_LENGTH = 6
+    MAX_PASSWORD_LENGTH = 512
+    MIN_EMAIL_LENGTH = 3
+    MAX_EMAIL_LENGTH = 255
+    MAX_SHARE_TOKEN_LENGTH = 255
+    MAX_GROUP_SHARE_ID_LENGTH = 255
+    MAX_HASHED_PASSWORD_LENGTH = 128
+
+    __tablename__ = "content_shares"
+
+    share_id = Column(
+        Integer, Sequence("seq__content_shares__share_id"), autoincrement=True, primary_key=True
+    )
+    content_id = Column(Integer, ForeignKey("content.id"), nullable=False)
+    author_id = Column(Integer, ForeignKey("users.user_id"), nullable=False)
+    email = Column(Unicode(MAX_EMAIL_LENGTH), nullable=False)
+    share_token = Column(Unicode(MAX_SHARE_TOKEN_LENGTH), nullable=False)
+    share_group_id = Column(Unicode(MAX_GROUP_SHARE_ID_LENGTH), nullable=False)
+    type = Column(Enum(ContentShareType), nullable=False)
+    _password = Column("password", Unicode(MAX_HASHED_PASSWORD_LENGTH), nullable=True)
+    enabled = Column(Boolean, unique=False, nullable=False, default=True)
+    created = Column(DateTime, unique=False, nullable=False, default=datetime.utcnow)
+    disabled = Column(DateTime, unique=False, nullable=True, default=None)
+    content = relationship("Content", remote_side=[Content.id], backref="shares")
+    author = relationship("User", remote_side=[User.user_id])
+
+    @classmethod
+    def _hash(cls, cleartext_password_or_token: str) -> str:
```

Comment:
Please add a docstring about what this method do

Reply:
done.